app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -1133,6 +1141,78 @@ User.reopenClass(Singleton, {
   },
 });
 
+// user status tracking
+User.reopen(Evented, {
```

Comment:
Instead of reopening the User class here, I wonder if it'll be better if we extract the responsibility into a `UserStatusTracking` class and pass the user object to it.

Reply:
> Anyway, I would prefer to leave it implemented this way for now, and improve it after I use it in more places, I'll understand better our use cases at that moment.

Sounds good :+1: Thank you for clarifying.